src/api/commits/compare_commit.rs

```diff
@@ -0,0 +1,73 @@
+use super::*;
+
+#[derive(serde::Serialize)]
+pub struct CompareCommitsBuilder<'octo, 'r> {
+    #[serde(skip)]
+    handler: &'r super::CommitHandler<'octo>,
+    base: String,
+    head: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    per_page: Option<u8>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    page: Option<u32>,
+}
+
+impl<'octo, 'r> CompareCommitsBuilder<'octo, 'r> {
+    pub(crate) fn new(
+        handler: &'r super::CommitHandler<'octo>,
+        base: String,
+        head: String,
+    ) -> Self {
+        Self {
+            handler,
+            base,
+            head,
+            page: None,
+            per_page: None,
+        }
+    }
+
+    /// Results per page (max 100).
+    pub fn per_page(mut self, per_page: impl Into<u8>) -> Self {
+        self.per_page = Some(per_page.into());
+        self
+    }
+
+    /// Page number of the results to fetch.
+    pub fn page(mut self, page: impl Into<u32>) -> Self {
+        self.page = Some(page.into());
+        self
+    }
+
+    /// Sends the actual request.
+    pub async fn send(self) -> crate::Result<models::commits::CommitComparison> {
```

Comment:
this endpoint accepts pagination params but doesn't return an array so I didn't wrap the return type in `Page`